packages/lesswrong/components/votes/VoteArrowIconSolid.tsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import classNames from 'classnames';
+import IconButton from '@material-ui/core/IconButton';
+import { SoftUpArrowIcon } from '../icons/softUpArrowIcon';
+import { SoftUpArrowIconCap } from '../icons/softUpArrowIconCap';
+import { useVoteColors } from './useVoteColors';
+import { registerComponent } from '@/lib/vulcan-lib';
+import { isEAForum } from '../../lib/instanceSettings';
+import type { BaseVoteArrowIconProps } from './VoteArrowIcon';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    color: theme.palette.grey[400],
+    fontSize: 'inherit',
+    width: 'initial',
+    height: 'initial',
+    padding: 0,
+    position: 'relative',
+    '&:hover': {
+      backgroundColor: 'transparent',
+    },
+  },
+  disabled: {
+    cursor: 'not-allowed',
+  },
+  smallArrow: {
+    opacity: isEAForum ? 0.7 : 0.6,
+    pointerEvents: 'none',
+  },
+  smallArrowLarge: {
+    opacity: isEAForum ? 0.7 : 0.6,
+    pointerEvents: 'none',
+  },
+  up: {
+  },
+  upLarge: {
+  },
+  down: {
+    transform: 'rotate(-180deg)',
+    top: 1
+  },
+  downLarge: {
+    transform: 'rotate(-180deg)',
+    top: 1
+  },
+  right: {
+    transform: 'rotate(-270deg)',
+  },
+  rightLarge: {
+    transform: 'rotate(-270deg)',
+  },
+  left: {
+    transform: 'rotate(-90deg)',
+  },
+  leftLarge: {
+    transform: 'rotate(-90deg)',
+  },
+  bigArrow: {
+    height: 10,
+    width: 10,
+    pointerEvents: 'none',
+    position: 'absolute',
+    top: '-70%',
+    fontSize: '82%',
+    opacity: 0,
+  },
+  bigArrowLarge: {
+    height: 14,
+    width: 14,
+    pointerEvents: 'none',
+    position: 'absolute',
+    top: '-90%',
+    fontSize: '100%',
+    opacity: 0,
+  },
+  bigArrowSolid: {
+    top: '-35%',
+    transform: 'scale(1.4)',
+  },
+  bigArrowSolidLarge: {
+    top: '-30%',
+    transform: 'scale(1.6)',
+  },
+  bigArrowCompleted: {
+    fontSize: '90%',
+    top: '-35%',
+    opacity: 1
+  },
+  bigArrowCompletedLarge: {
+    fontSize: '110%',
+    top: '-35%',
+    opacity: 1
+  },
+  entering: {
+    transition: `opacity ${theme.voting.strongVoteDelay}ms cubic-bezier(0.74, -0.01, 1, 1) 0ms`,
+  }
+});
+
+const VoteArrowIconSolid = ({
+  orientation,
+  enabled = true,
+  color,
+  voted,
+  eventHandlers,
+  bigVotingTransition,
+  bigVoted,
+  bigVoteCompleted,
+  alwaysColored,
+  strongVoteDelay,
+  largeArrow = false,
+  classes,
+}: BaseVoteArrowIconProps & {
+  classes: ClassesType
+}) => {
+
+  const { mainColor, lightColor } = useVoteColors(color);
+
+  const iconSize = largeArrow ? 14 : 10;
+
+  const Icon = (
+    <SoftUpArrowIcon
+      style={{
+        color: voted || alwaysColored ? mainColor : 'inherit',
+        height: iconSize,
+        width: iconSize,
```

Comment:
Is this refactored from existing code?  Seems like it could just be using our regular styling system (except maybe the color thing)

Reply:
Yeah, refactored from existing